ui/components/component-library/tag-url/tag-url.types.ts

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import type {
+  PolymorphicComponentPropWithRef,
+  StyleUtilityProps,
+} from '../box';
+import { AvatarFaviconProps } from '../avatar-favicon';
+import { IconProps } from '../icon';
+import { TextProps } from '../text';
+import { ButtonLinkProps } from '../button-link';
+
+export interface TagUrlStyleUtilityProps extends StyleUtilityProps {
+  /**
+   * The src accepts the string of the image to be rendered
+   */
+  src?: string;
+  /**
+   * The showLockIcon accepts a boolean prop to render the lock icon instead of https in label
+   */
+  showLockIcon?: boolean;
+  /**
+   * It accepts all the props from Avatar Favicon
+   */
+  avatarFaviconProps?: AvatarFaviconProps<'span'>;
+  /**
+   * It accepts all the props from Icon
+   */
+  lockIconProps?: IconProps<'span'>;
+  /**
+   * The text content of the TagUrl component
+   */
+  label: string;
+  /**
+   * It accepts all the props from Text Component
+   */
+  labelProps?: TextProps<'p'>;
+  /**
+   * If we want a button in TagUrl component.
+   */
+  actionButtonLabel?: string | React.ReactNode;
+  /**
+   * It accepts all the props from ButtonLink
+   */
+  actionButtonProps?: ButtonLinkProps<'button'>;
```

Comment:
Hmm I'm not sure how this component will act if we pass it an href or attributes related to an `a` tag. I've created some stories that use href and they seem to be working without issue so we can monitor it.